subsys/caf/modules/ble_adv.c

```diff
@@ -598,6 +676,37 @@ static void fast_adv_end_fn(struct k_work *work)
 	__ASSERT_NO_MSG(!fast_adv);
 }
 
+static void rpa_rotate_fn(struct k_work *work)
+{
+	ARG_UNUSED(work);
+
+	if (IS_ENABLED(CONFIG_CAF_BLE_ADV_GRACE_PERIOD) && (state == STATE_GRACE_PERIOD)) {
+		(void)k_work_cancel_delayable(&grace_period_end);
```

Comment:
Why do you think so?

We can have the RPA rotation in the Grace Period state. In that case, we shut down the grace period advertising prematurely.

Reply:
`grace_period_end_fn` triggers `update_state(STATE_OFF)` which eventually cancels the work. So that there is no need to cancel the work manually.
In general I think we should avoid updating works related to given state manually (unless necessary  - e.g. RPA rotation). Relying only on state changes simplifies understanding module's behavior.

Reply:
Can we rely on the `grace_period_end_fn` to cancel the work here?

Reply:
Yes, that's right. Sorry, I didn't get what you mean with your first comment